nexus/src/db/datastore.rs

```diff
@@ -1938,6 +1943,89 @@ impl DataStore {
                 ))
             })
     }
+
+    pub async fn users_builtin_list_by_name(
+        &self,
+        opctx: &OpContext,
+        pagparams: &DataPageParams<'_, Name>,
+    ) -> ListResultVec<UserBuiltin> {
+        use db::schema::user_builtin::dsl;
+        opctx.authorize(authz::Action::ListChildren, authz::FLEET)?;
+        paginated(dsl::user_builtin, dsl::name, pagparams)
+            .select(UserBuiltin::as_select())
+            .load_async::<UserBuiltin>(self.pool_authorized(opctx)?)
+            .await
+            .map_err(|e| {
+                public_error_from_diesel_pool(
+                    e,
+                    ResourceType::User,
+                    LookupType::Other("Listing All".to_string()),
+                )
+            })
+    }
+
+    pub async fn user_builtin_fetch(
+        &self,
+        opctx: &OpContext,
+        name: &Name,
+    ) -> LookupResult<UserBuiltin> {
+        use db::schema::user_builtin::dsl;
+        opctx.authorize(authz::Action::ListChildren, authz::FLEET)?;
```

Comment:
Should this be Action::Read?

Reply:
I don't think so?  Check out https://github.com/oxidecomputer/omicron/blob/038dbd04fbc292b5f7880c0d0c723ef182145cb2/nexus/src/authz/omicron.polar#L41-L80.  I chose to treat these as children of the top-level Fleet.  Let me know what you think.

Reply:
From these descriptions it still sounds like `Read` to me? Unless we consider resolving a single object by name to be a listing operation

Reply:
Oh, I see your point.

What resource would the "read" action be applied to?  I assumed at first that you meant that we'd create a new Polar resource for UserBuiltin as a child of Fleet.  That's totally reasonable.  I elected not do that only because it's a bunch more plumbing and it wasn't clear to me that it was useful to be able to assign permissions on builtin users.  Admittedly, it's kind of implicit to piggy-back on the fleet-level list-children permission.

The policy currently doesn't say anything about most system-wide resources, including built-in users and all the hardware resources that we'll eventually have.  Until we have reason to think that people are going to want to split these up, I'd propose treating these similar to `ProjectResource` today: we could create one Polar type for `SystemResource` with no roles of its own, but permissions granted based on one's roles on the Fleet.  It's a little hollow because unlike `ProjectResource`, there's only one possible parent (so there would be no state in a `SystemResource` struct).  But it might be more explicit.  What do you think?

Another option is to keep the resource in this check as `Fleet`, treating the users and other system-wide information as part of the Fleet itself.  I think that's not great because I can imagine even pretty unprivileged users getting "read" permission on the Fleet to be able to view system-wide configuration or something, but that doesn't mean they should be able to see _all_ system-wide resources.

Reply:
I like the middle ground of creating one `SystemResource` for all these things to share. I agree with Tess than using a list children permission to read a single thing is weird.

Reply:
Done in 9f562e95 -- let me know what you think.